simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -12,320 +14,76 @@ def _alpha(src):
+        Locations where the two fields are measured.
+        If it is a 3D array, `locations[0]` are the locations of the first field
+        measurements and `locations[1]` are the locations of the second field measurements.
+        Otherwise, they are assumed to be at the same location for a 2D array input.
     """
 
-    def __init__(
-        self,
-        locations=None,
-        locations_e=None,
-        locations_h=None,
-    ):
-        # check if locations_e or h have been provided
-        if (locations_e is not None) and (locations_h is not None):
-            # check that locations are same size
-            if locations_e.size == locations_h.size:
-                self._locations_e = locations_e
-                self._locations_h = locations_h
-            else:
-                raise ValueError("location h needs to be same size as location e")
-
-            locations = np.hstack([locations_e, locations_h])
-        elif locations is not None:
-            # check shape of locations
-            if isinstance(locations, list):
-                if len(locations) == 2:
-                    self._locations_e = locations[0]
-                    self._locations_h = locations[1]
-                elif len(locations) == 1:
-                    self._locations_e = locations[0]
-                    self._locations_h = locations[0]
-                else:
-                    raise ValueError("incorrect size of list, must be length of 1 or 2")
-                locations = locations[0]
-            elif isinstance(locations, np.ndarray):
-                self._locations_e = locations
-                self._locations_h = locations
-            else:
-                raise ValueError("locations need to be either a list or numpy array")
-        else:
-            locations = np.array([[0.0]])
-
-        super().__init__(locations)
-
-    @property
-    def locations_e(self):
-        """Locations for the electric field measurements for each datum.
-
-        Returns
-        -------
-        (n_data, dim) numpy.ndarray
-            Location for the electric field measurements for each datum.
-        """
-        return self._locations_e
-
-    @property
-    def locations_h(self):
-        """Locations for the magnetic field measurements for each datum.
-
-        Returns
-        -------
-        (n_data, dim) numpy.ndarray
-            Location for the magnetic field measurements for each datum.
-        """
-        return self._locations_h
+    _loc_names = ("First", "Second")
 
-    def _eval_apparent_conductivity(self, src, mesh, f):
-        if mesh.dim < 3:
-            raise NotImplementedError(
-                "ApparentConductivity receiver not implemented for dim < 3."
+    def __init__(self, locations1, locations2, **kwargs):
+        if "locations" in kwargs:
+            raise TypeError(
+                f"locations is not a valid argument for the {type(self)} class."
             )
```

Comment:
This is the error raising I mentioned in my review. This is only needed if we allow `kwargs` to be passed to the constructor, and we now need it only because the deprecated classes are ultimately children of this new class.

Reply:
It’s there to guard against passing `locations` as a keyword argument to the new classes. Once those are switched to not accept `**kwargs` it can be removed.